### ISSUE TITLE
Failed to save: Cannot read property 'buffer' of undefined

### ISSUE THREAD

**Issue by grudloff** [microsoft/vscode-jupyter#7435]
## Environment data

-   VS Code version: 1.60.0
-   Jupyter Extension version: v2021.8.2031190714
-   Python Extension version: v2021.9.1191016588
-   OS and version: Windows_NT x64 10.0.19043
-   Python and/or Anaconda version: python 3.9.5 & conda 4.10.3
-   Type of virtual environment used : conda
-   Jupyter server running: Local

## Problem

Unable to save a jupyter notebook, I get a popup stating that:

> Failed to save 'filename.ipynb': Cannot read property 'buffer' of undefined

This issue has been repeated several times since the last update from vscode, the only option is to save important data on another file, closing and then opening again. Now it happened again as I tried to save, it undid all the changes I had made since the last save, fortunately, I could redo everything from 'Edit > Redo' menu.

**Comment by IanMatthewHuff** [comment]
@grudloff Is this happening with one specific file? Or any file that you have on the system? If it's one file, could you provide it so that I could take a peek if we are having an issue specific to it?

**Comment by DonJayamanne** [comment]
Duplicate of https://github.com/microsoft/vscode/issues/131914

**Comment by greazer** [comment]
@grudloff, This issue has been tricky to reproduce. We do have issue #131914, but it doesn't involve jupyter notebook files. So for now I'd like to keep this issue open and if you can help us find some repro steps as well as answer @IanMatthewHuff's questions as best you can, that would be awesome.

**Comment by greazer** [comment]
A few more questions:
- Are you manually saving with Ctrl+S or File->Save? I assume so.
- Is there a cell running in your notebook when you try to save or are all cells stopped (kernel in idle state)?
- Shot in the dark: If you restart your kernel before saving, does that resolve the situation? 

Really sorry this is happening to you. Hoping to get to the bottom of it soon!

**Comment by grudloff** [comment]
Hello @IanMatthewHuff @DonJayamanne @greazer , thanks for the support. It seems this issue is a dupe of [microsoft/vscode#131914](https://github.com/microsoft/vscode/issues/131914). Anyways, I was unable to save this file until vscode crashed and for some reason, after that, I was able to do it without any problem. I haven't had any similar issues again since then. I am going to answer your questions for completeness in case it is useful to fix this issue.

> @grudloff Is this happening with one specific file? Or any file that you have on the system? If it's one file, could you provide it so that I could take a peek if we are having an issue specific to it?

Just happening in one file. I can't provide the file, but it was the training of a tensorflow conv net with the use of optuna for hyperparameter tuning. So, very large running cell times, and very large cell outputs.

> A few more questions:
> 
> * Are you manually saving with Ctrl+S or File->Save? I assume so.
> * Is there a cell running in your notebook when you try to save or are all cells stopped (kernel in idle state)?
> * Shot in the dark: If you restart your kernel before saving, does that resolve the situation?
> 
> Really sorry this is happening to you. Hoping to get to the bottom of it soon!

* I tried both, with the same results.
* First they were running and I tried to save and it didn't work, then with no kernel, same issue.
* Restarting the kernel did nothing to change the situation.
